src/audio/wav64.c

```diff
@@ -148,8 +150,23 @@ static inline void rsp_vadpcm_decompress(void *input, int16_t *output, bool ster
 
 #endif /* VADPCM_REFERENCE_DECODER */
 
-void raw_waveform_read(samplebuffer_t *sbuf, int base_rom_addr, int wpos, int wlen, int bps) {
-	uint32_t rom_addr = base_rom_addr + (wpos << bps);
+void raw_waveform_read(samplebuffer_t *sbuf, int currentFile, int wpos, int wlen, int bps) {
+	//uint32_t rom_addr = base_rom_addr + (wpos << bps);
+	uint8_t* ram_addr = (uint8_t*)samplebuffer_append(sbuf, wlen);
+	int bytes = wlen << bps;
+
+	uint32_t t0 = TICKS_READ();
+	// Run the DMA transfer. We rely on libdragon's PI DMA function which works
+	// also for misaligned addresses and odd lengths.
+	// The mixer/samplebuffer guarantees that ROM/RAM addresses are always
+	// on the same 2-byte phase, as the only requirement of dma_read.
+	read(currentFile, ram_addr, bytes);
+	lseek(currentFile, -bytes, SEEK_CUR);
```

Comment:
I can't understand how it works if you're seeking back, and ignoring `wpos` at the same time. I suggest you test the implementation with `rom:/` since `pi:/` is pretty new too.